nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/FetchSolr.java

```diff
@@ -0,0 +1,432 @@
+    static {
+        SUPPORTED_SEARCH_COMPONENTS.addAll(Arrays.asList(StatsParams.STATS, FacetParams.FACET));
+    }
+    public static final Set<String> SEARCH_COMPONENTS_ON = new HashSet<String>();
+    static {
+        SEARCH_COMPONENTS_ON.addAll(Arrays.asList("true", "on", "yes"));
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final ComponentLog logger = getLogger();
+
+        FlowFile flowFileRequest = session.get();
+        boolean keepOriginal;
+
+        if (flowFileRequest == null) {
+            if (context.hasNonLoopConnection()) {
+                return;
+            }
+            flowFileRequest = session.create();
+            keepOriginal = false;
+        } else {
+            keepOriginal = true;
+        }
+
+        final String queryString = context.getProperty(SOLR_QUERY_STRING).evaluateAttributeExpressions(flowFileRequest).getValue();
+
+        final Map<String,String[]> solrParams = SolrRequestParsers.parseQueryString(queryString).getMap();
+        final Set<String> searchComponents = extractSearchComponents(solrParams);
+        final SolrQuery solrQuery = new SolrQuery();
+        solrQuery.add(new MultiMapSolrParams(solrParams));
+
+        final String requestHandler = context.getProperty(REQUEST_HANDLER).getValue();
+        solrQuery.setRequestHandler(requestHandler);
+
+        final QueryRequest req = new QueryRequest(solrQuery);
+
+        if (isBasicAuthEnabled()) {
+            req.setBasicAuthCredentials(getUsername(), getPassword());
+        }
+
+        flowFileRequest = session.putAttribute(flowFileRequest, ATTRIBUTE_SOLR_CONNECT, getSolrLocation());
+        if (SOLR_TYPE_CLOUD.equals(context.getProperty(SOLR_TYPE).getValue())) {
+            flowFileRequest = session.putAttribute(flowFileRequest, ATTRIBUTE_SOLR_COLLECTION, context.getProperty(COLLECTION).evaluateAttributeExpressions().getValue());
+        }
+        flowFileRequest = session.putAttribute(flowFileRequest, ATTRIBUTE_SOLR_QUERY, solrQuery.toString());
+
+        FlowFile flowFileResponse;
+        if (!keepOriginal) {
+            flowFileResponse = flowFileRequest;
+            flowFileRequest = null;
+        } else {
+            flowFileResponse = session.create(flowFileRequest);
+        }
+
+        try {
+            final QueryResponse response = req.process(getSolrClient());
+
+            Map<String,String> responseAttributes = new HashMap<>();
+            responseAttributes.put(ATTRIBUTE_CURSOR_MARK, response.getNextCursorMark());
+            responseAttributes.put(ATTRIBUTE_SOLR_STATUS, String.valueOf(response.getStatus()));
+            responseAttributes.put(ATTRIBUTE_QUERY_TIME, String.valueOf(response.getQTime()));
+            flowFileResponse = session.putAllAttributes(flowFileResponse, responseAttributes);
+
+            if (response.getResults().size() > 0) {

```

Comment:
What is the expected behavior when there are more results than were asked for in the initial search?

Say there are 1k total results, but the query asked for rows 0 to 10.

Are we going to page through the results and send out multiple flow files? 

Do we just don't handle this case and it is up to the user to make the rows large enough (which also means they could blow up memory if they ask for too many results in a single query)?